HOOMD-blue 2.9.7: HPCM compilation fails

## Description

Compilation fails when HPCM is enabled.


## Script

```
-DSINGLE_PRECISION=OFF -DENABLE_MPI=ON -DENABLE_MPI_CUDA=ON -DENABLE_CUDA=ON -DCUDA_ARCH_LIST=80 -DBUILD_CGCMM=ON -DBUILD_DEPRECATED=ON -DBUILD_HPMC=ON -DBUILD_MD=ON -DBUILD_METAL=ON -DBUILD_DEM=ON -DBUILD_TESTING=ON -DBUILD_VALIDATION=OFF -DENABLE_HPMC_MIXED_PRECISION=OFF -DENABLE_HOST=ON
```

<!-- Attach any input files needed to execute the script. -->

## Output

```
/dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/hoomd-blue/hoomd/hpmc/ShapePolyhedron.h(594): error: no instance of overloaded function "rotate" matches the argument list
            argument types are: (quat<double>, vec3<float>)

1 error detected in the compilation of "/dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/hoomd-blue/hoomd/hpmc/all_kernels_polyhedron.cu".
-- Removing /dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/easybuild_obj/hoomd/hpmc/CMakeFiles/cuda_compile_1.dir//./cuda_compile_1_generated_all_kernels_polyhedron.cu.o
/mnt/tier2/apps/USE/easybuild/staging/2021.1/software/CMake/3.20.1-GCCcore-10.3.0/bin/cmake -E rm -f /dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/easybuild_obj/hoomd/hpmc/CMakeFiles/cuda_compile_1.dir//./cuda_compile_1_generated_all_kernels_polyhedron.cu.o
CMake Error at cuda_compile_1_generated_all_kernels_polyhedron.cu.o.Release.cmake:280 (message):
  Error generating file
  /dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/easybuild_obj/hoomd/hpmc/CMakeFiles/cuda_compile_1.dir//./cuda_compile_1_generated_all_kernels_polyhedron.cu.o


make[2]: *** [hoomd/hpmc/CMakeFiles/_hpmc.dir/build.make:122: hoomd/hpmc/CMakeFiles/cuda_compile_1.dir/cuda_compile_1_generated_all_kernels_polyhedron.cu.o] Error 1
make[2]: Leaving directory '/dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/easybuild_obj'
make[1]: *** [CMakeFiles/Makefile2:2695: hoomd/hpmc/CMakeFiles/_hpmc.dir/all] Error 2
make[1]: Leaving directory '/dev/shm/hoomdblue/2.9.7/foss-2021a-CUDA-11.3.1/easybuild_obj'
make: *** [Makefile:149: all] Error 2
```

## Expected output

I expected it to compile until the end.

## Configuration

<!-- What is your system configuration? -->

<!-- Remove items that do not apply. -->

Platform:
- Linux: Rocky 8.5
- GPU: A100, CUDA 11.3.1
- CPU: AMD Rome 2x64c

Installation method:
- Compiled from source

<!-- What software versions do you have? -->

Versions:
- Python version: 3.9.5
- **HOOMD-blue** version: 2.9.7
- CUDA 11.3.1
- CMake 3.20.1

## Developer

<!-- Who should implement the fix? We would welcome your contribution! -->

--- a/hoomd/hpmc/ShapePolyhedron.h
+++ b/hoomd/hpmc/ShapePolyhedron.h
@@ -590,7 +590,7 @@ DEVICE inline bool test_narrow_phase_overlap( vec3<OverlapReal> dr,
             for (unsigned int ivert = 0; ivert < 3; ++ivert)
                 {
                 unsigned int idx_a = a.data.face_verts[offs_a+ivert];
-                vec3<float> v = a.data.verts[idx_a];
+                vec3<OverlapReal> v = a.data.verts[idx_a];
                 v = rotate(q,v) + dr;
                 U[ivert][0] = v.x; U[ivert][1] = v.y; U[ivert][2] = v.z;
                 }

